pkg/provisioner/ironic/ironic.go

```diff
@@ -730,6 +731,53 @@ func (p *ironicProvisioner) getUpdateOptsForNode(ironicNode *nodes.Node) (update
 	return updates, nil
 }
 
+// makeHintMap converts a RootDeviceHints instance into a string map
+// suitable to pass to ironic.
+func makeHintMap(source *metal3v1alpha1.RootDeviceHints) map[string]string {
```

Comment:
Could we make this public (and put it elsewhere)? I realize OpenShift isn't a primary concern of Metal3 itself, but maybe having a general policy of things that convert BareMetalHost structs -> raw ironic data be library functions would be consistent with how we did HardwareDetails.

Reply:
Good point, this is likely to be useful in the openshift installer code. Maybe all of the new stuff should go into a new package? Something like `pkg/devicehints`?

Reply:
I moved it to `pkg/provisioner/ironic/devicehints`.